### DISCUSSION TITLE
Latest preview

### DISCUSSION THREAD

**Discussion by IeuanWalker** [CommunityToolkit/Maui#272] [General]
Should i be able to use the converters in the latest preview?
```
Severity	Code	Description	Project	File	Line	Suppression State
Error	XLS0414	The type 'mct:IsNotNullOrEmptyConverter' was not found. Verify that you are not missing an assembly reference and that all referenced assemblies have been built.

```
![image](https://user-images.githubusercontent.com/6544051/152135275-c5c07b3b-9d6c-443a-a07b-f7eb8ef8da13.png)

Also getting a lot of warnings -
![image](https://user-images.githubusercontent.com/6544051/152135494-5ca58c98-191d-48af-b975-263a167c5053.png)

**Comment by bijington** [comment]
@IeuanWalker you should but I made a typo on the namespace declaration 😞. It is fixed but hasn't been released yet: #263 

In the meantime you can change your definition to:

`xmlns:mct="http://schemas.microsoft.com/dotnet/2022/maui/tookit"` NOTE the lack of the L in toolkit 🤦 

Really sorry about this

**Reply by IeuanWalker** [reply]
@bijington no problem thanks.

That solved the xaml error, but now getting a build error -
![image](https://user-images.githubusercontent.com/6544051/152143452-fca3595d-9979-49d5-a712-35ef10c9322f.png)

I've done a clean/ rebuild

**Reply by bijington** [reply]
Hmm that is very odd! Last I checked our samples app was building with this exact converter fine: https://github.com/CommunityToolkit/Maui/blob/main/samples/CommunityToolkit.Maui.Sample/Pages/Converters/IsNotNullOrEmptyConverterPage.xaml

Just to double check are you building with Visual Studio 2022 Preview 5? https://docs.microsoft.com/en-gb/visualstudio/releases/2022/release-notes-preview#1710-pre50--visual-studio-2022-version-171-preview-5-newreleasebutton

**Reply by IeuanWalker** [reply]
@bijington Ye latest version - 
![image](https://user-images.githubusercontent.com/6544051/152174011-5061ad87-1b08-483f-bce9-b4178e392036.png)

Build fails just adding the resource to the page, no xaml errors though - 
```xaml
    <ContentPage.Resources>
        <ResourceDictionary>
            <mct:IsNotNullOrEmptyConverter x:Key="IsNotNullOrEmptyConverter" />
        </ResourceDictionary>
    </ContentPage.Resources>
```

**Reply by VladislavAntonyuk** [reply]
Does it work if you use clr-namespace and assembly (like you do with viewModels)?

**Reply by IeuanWalker** [reply]
@VladislavAntonyuk Not sure if I'm doing it right but this is the only namespace that intellisense found, and it doesn't find the converter - 
![image](https://user-images.githubusercontent.com/6544051/152177164-6a76e64b-b217-4578-80d8-c9f88eebf14c.png)

**Reply by VladislavAntonyuk** [reply]
try this namespace: CommunityToolkit.Maui.Converters

**Reply by IeuanWalker** [reply]
@VladislavAntonyuk thanks, that works -
![image](https://user-images.githubusercontent.com/6544051/152213312-b60487de-0290-4bec-9bb0-0c2f74ef916e.png)

**Reply by VladislavAntonyuk** [reply]
but it's a workaround rather than a solution. With the next preview, you should be able to use `xmlns:mct="http://schemas.microsoft.com/dotnet/2022/maui/toolkit"`

**Reply by bijington** [reply]
@IeuanWalker I have been able to reproduce the issue that you are seeing locally. The fix for me was to add this line wherever you create your maui app:

```csharp
builder.UseMauiApp<App>().UseMauiCommunityToolkit();
```

Are you able to confirm whether you are only using the toolkit inside XAML?

**Reply by IeuanWalker** [reply]
@bijington ye your right that fixes it.

I do get this warning now though - 
![image](https://user-images.githubusercontent.com/6544051/152239847-b3b5173e-f35c-40f7-92b9-74e2c2fdc3f0.png)

My target is iOS 11 and i don't support windows app - 
![image](https://user-images.githubusercontent.com/6544051/152239993-db2a9a9a-fc07-4209-ac18-30daa3829383.png)

**Reply by TheCodeTraveler** [reply]
Thanks @IeuanWalker! I added this to an Issue that we have open to address these warnings: 
https://github.com/CommunityToolkit/Maui/issues/268

**Reply by bijington** [reply]
Well at least we are dropping in severity 😄. I also see that issue but I have a couple of Windows related setup issues on my machine. I had put it down to that but it sounds like it may be a general issue with the package then. I see the info has already been added to #268 as I was typing this

**Reply by IeuanWalker** [reply]
Thanks for all your help guys, will keep an eye on that issue.

**Comment by CliffAgius** [comment]
Closed as per the Community Stand-up discussion here: https://youtu.be/N9wMcBP4jtg?t=2889
